### PR TITLE
Final changes to taar_lite_guid_guid PR

### PR DESCRIPTION
**Pull request by crankycoder**
This corrects the format of the expected output of .key_all() function.

* The ETL job has also been explicitly split into extract, transform and load functions.
* taar_utils.store_json_to_s3 is now being reused as that was just duplicate code

### CODE REVIEW

File: mozetl/taar/taar_lite_guidguid.py [hunk 1]
```diff
@@ -241,8 +187,26 @@ def main(date, bucket, prefix):
             value_json[_id] = n
         result_json[key_addon] = value_json
```
Comment (acmiyaguchi) [inline mozetl/taar/taar_lite_guidguid.py]: Is this missing the return value?

Comment (acmiyaguchi) [reply mozetl/taar/taar_lite_guidguid.py]: I like structuring these jobs in the following way:
```
extract :: IO -> DataFrame
transform :: DataFrame -> DataFrame
load :: DataFrame -> IO
```

The main advantage is having a consistent container type for testing the transformation method, which should do the bulk of the work. Additionally, IO can be mocked easily in this repository through the use of the boto3 `@mock_s3` decorator, which redirects writes to the filesystem.

File: tests/test_taar_lite_guidguid.py [hunk 2]
```diff
@@ -23,17 +23,31 @@
...
-        {"test-guid-2": 1,
-         "test-guid-3": 2,
-         "test-guid-4": 2
-         },
-    "test-guid-2":
-        {"test-guid-1": 2,
-         "test-guid-5": 1,
-         "test-guid-6": 1
-         }}
+MOCK_ADDON_INSTALLATIONS = [
```
Comment (acmiyaguchi) [inline tests/test_taar_lite_guidguid.py]: A better name might be `EXPECTED_ADDON_INSTALLATIONS`, since mocking is more generally faking data from sources outside of the script's control. You may also consider making this a dataframe, and using the `df_equals` test-fixture found in `conftest.py`. Example usage in this repo:

https://github.com/mozilla/python_mozetl/blob/054b34e93f687d344c1b3e9ab613681166858ef1/tests/engagement/test_daily_retention.py#L69-L85

File: tests/test_taar_lite_guidguid.py [hunk 3]
```diff
@@ -47,12 +61,12 @@
...
             return_value=MOCK_TELEMETRY_SAMPLE)
 @mock_s3
-def test_load_training_from_telemetry(spark):
+def test_extract_telemetry(spark):
     # Sanity check that mocking is happening correctly.
-    assert taar_lite_guidguid.load_training_from_telemetry(spark) == MOCK_TELEMETRY_SAMPLE
+    assert taar_lite_guidguid.extract_telemetry(spark) == MOCK_TELEMETRY_SAMPLE
 
 # Exercise the only part of the ETL job happening outside of spark.
 def test_addon_keying():
```
Comment (acmiyaguchi) [inline tests/test_taar_lite_guidguid.py]: You should test the output of the transformation method, perhaps using the `df_equals` test-fixture found in `conftest.py`. Example usage in this repo:

https://github.com/mozilla/python_mozetl/blob/054b34e93f687d344c1b3e9ab613681166858ef1/tests/engagement/test_daily_retention.py#L69-L85

Testing the transformation should subsume this unit test. I've been starting to focus more on testing correctness of the API boundary (`transform`), but sanity checks for implementation details like this one help too.
